client/src/components/advancedSelectWidget/MultiTypeAdvancedSelectComponent.js

```diff
@@ -148,6 +148,37 @@ const MultiTypeAdvancedSelectComponent = ({
     ? AdvancedMultiSelect
     : AdvancedSingleSelect
   const extraSelectProps = isMultiSelect ? {} : { showRemoveButton: false }
+
+  /**
+   * Generates field filters with respect to the dictionary. If there is
+   * no filter(s) to be applied, returns the default value.
+   * @returns {Object} Holds the filter values to be applied to the field.
+   */
+  const generateLocationFilters = useCallback(() => {
+    // Create a default empty object.
+    const locationFilters = Object.create({})
+    const currentFilter = filters[0]
+    if (
+      currentFilter !== undefined &&
+      currentFilter[entityType] &&
+      currentFilter[entityType].typeFilter
```

Comment:
Can be simplified using optional chaining: `currentFilter?.[entityType]?.typeFilter`